### ISSUE TITLE
model_restart test (seemingly randomly) failing on weaver after 1433 got merged

### ISSUE THREAD

**Issue by bartgol** [E3SM-Project/scream#1453]
What I have so far:

- The test seems to have somewhat random failures, but I observed (though after running 1k times memories are mixing) that the test passes the first time it's run right after cmake ran, while subsequent calls to ctest exibit failures.
- Saving the output file `model_output_baseline.***.nc` after the 1st run after cmake, and after a subsequent run, shows sizeable diffs in almost all fields. This suggests that some input file is modified (though I didn't see any diffs pre/post 1st run), or that the presence of some files trigger different behavior in the executable. Recall, the test should be DETERMINISTIC, since it uses fixed (i.e., no random) inputs.
- The test fails also without ctest parallel execution.
- Reverting master to before 1433 was merged makes the test consistently pass
- While I don't know if this is part of the problem, the kokkos CXX flags (as reported by kokkoscore.dir/flags.make) are different pre/post 1433 merge. In particular, before that PR, our debug bfb flags are not present, while they are present afterword. That is
```
$ cat /path/to/pre-merge/build/kokkoscore.dir/flags | grep CXX_FLAGS
CXX_FLAGS = -g -fopenmp -march=znver2 -mtune=znver2 -w -std=gnu++14
$ cat /path/to/post-merge/build/kokkoscore.dir/flags | grep CXX_FLAGS
CXX_FLAGS =  -ffp-contract=off -Wall -g -g -O0 -fopenmp-simd -fopenmp -march=znver2 -mtune=znver2 -w -std=gnu++14
```

**Comment by bartgol** [comment]
Forget about ctest. From the model_restart folder, 

```
$ ./model_restart --ekat-test-params ifile=input_baseline.yaml
$ ./model_restart --ekat-test-params ifile=input_baseline.yaml
```
the second baseline output nc file matches the first. However
```
$ ./model_restart --ekat-test-params ifile=input_baseline.yaml
$ ./model_restart --ekat-test-params ifile=input_initial.yaml
$ ./model_restart --ekat-test-params ifile=input_baseline.yaml
```
the second baseline output nc file is different from the first one.

**Comment by bartgol** [comment]
Even more bizarre! Run `./model_restart --ekat-test-params ifile=input_baseline.yaml`, and save the `model_output_baseline.blah.nc` file. Run again, and again, and again. Eventually (took me 3 runs) the file diffs. Save the new file too. Every output of every subsequent run matches one of the two previously saved files. If you run a few times, but always get the same output file, try running with `ifile=input_initial.yaml` once, then go back to baseline runs; that seems to contribute to switching the output as well.

I checked the time stamp of the files in the bld folder, and the only ones getting updated are `mass.out` (which is a diagnostic file written by homme), the `io_perf_summary_x.y` files written by scorpio, and the `fort.98` file, which contains the line 
```
    P3_INIT (reading/creating look-up tables) ...
```
And, of course, the `model_output_baseline.xyz.nc` output file. None of these files should have any effect on the calculations.

All of this seems to point to something like a bool var that is not initialized correctly (or not at all), so that the execution switches between two possible outcomes.

**Comment by bartgol** [comment]
I think the error was due to a missing entry in the input namelist file. After adding it, the test started passing for me.

This might go away with #1411. I will start working on that one, to verify it, and if the test passes in that branch, I'll close this.

**Comment by bartgol** [comment]
Logging here some experiments I did.

I grabbed a vanilla version of current master (f954b54), and ran `ctest -R model_baseline` ten times. After each run, I renamed the `model_output_baseline.*.nc` file to `$run_number.nc` for the sake of comparison later, but other than that, I left all the generated files there (e.g., p3 tables). The ten runs generated only two possible outputs. In particular,
- 1.nc == 5.nc == 7.nc
- 2.nc == 3.nc == 4.nc == 6.nc == 8.nc == 9.nc == 10.nc

After this, I modified `src/physics/rrtmgp/CMakeFiles/rrtmgp.dir/flags.make`, and added `--fmad=false` to the CXX flags. Then did ten runs, like before. The generated outputs were all the same, though they didn't match neither 1.nc nor 2.nc (understandably, since floating point calculations changed).
